packages/dds/tree/src/simple-tree/api/treeNodeApi.ts

```diff
@@ -167,7 +168,44 @@ export const treeNodeApi: TreeNodeApi = {
 		eventName: K,
 		listener: TreeChangeEvents[K],
 	): Off {
-		return getKernel(node).on(eventName, listener);
+		const kernel = getKernel(node);
+		if (kernel.hydrated === undefined) {
+			throw new UsageError(
+				"Subscribing to events for a node which is Unhydrated is unsupported",
```

Comment:
Looks like we currently have support for subscribing to unhydrated nodes, which fail to give events while unhydrated, but then start reporting when hydrated.

This seems like both a feature and an undocumented bug. We probably need to preserve that feature, so a different implementation strategy is needed here.

I think I may give the node kernel a hydrated event, or a after hydrate method or something like that.

Reply:
Fixed.